### ISSUE TITLE
Stateful lstm questions regarding state resetting and timestep invariance

### ISSUE THREAD

**Issue by surGeonGG** [keras-team/keras#10231]
I have two questions.

Question 1: Let's say my program calls predict() and then calls predict() again later (as new data has come in). 

Has the state been reset between the predicts? If not, does the state also persist between a fit() and a predict() call?

Question 2: In a stateful lstm and assuming we have one long sequence that is subdivided into multiple samples that are passed in one by one with a batch_input_shape=(1, 200, features). 

Assuming many batches have already been passed in (meaning there's state from previous batches), does the time distributed prediction of timesteps depend on where in the sample a timestep is located?

In other words, will the prediction of a timestep be the same whether it is the first or last timestep in a sample given the previous 10 000 timesteps in the combined long sequence are the same?

Thanks.
